app/vmctl/influx/v2/adapter.go

```diff
@@ -0,0 +1,110 @@
+package v2
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	influxdb2 "github.com/influxdata/influxdb-client-go/v2"
+	"github.com/influxdata/influxdb-client-go/v2/api"
+)
+
+type Config struct {
+	Addr   string
+	Token  string
+	Org    string
+	Bucket string
+
+	Filter    Filter
+	TLSConfig *tls.Config
+}
+
+// Filter contains configuration for filtering
+// the timeseries
+type Filter struct {
+	TimeStart string
+	TimeEnd   string
+}
+
+type Series struct {
+	Measurement string
+	Field       string
+}
+
+type Adapter struct {
+	client   influxdb2.Client
+	queryAPI api.QueryAPI
+	bucket   string
+	start    string
+	stop     string
+}
+
+func (a *Adapter) Bucket() string {
+	return a.bucket
+}
+
+func NewAdapter(cfg Config) *Adapter {
+	client := influxdb2.NewClientWithOptions(
+		cfg.Addr,
+		cfg.Token,
+		influxdb2.
+			DefaultOptions().
+			SetTLSConfig(cfg.TLSConfig),
+	)
+	adapter := &Adapter{
+		client:   client,
+		queryAPI: client.QueryAPI(cfg.Org),
+		bucket:   cfg.Bucket,
+		start:    "-inf",
+		stop:     "now()",
+	}
+	if cfg.Filter.TimeStart != "" {
+		adapter.start = fmt.Sprintf("%s", cfg.Filter.TimeStart)
+	}
+	if cfg.Filter.TimeEnd != "" {
+		adapter.stop = fmt.Sprintf("%s", cfg.Filter.TimeEnd)
+	}
+	return adapter
+}
+
+func (a *Adapter) Explore(ctx context.Context) ([]Series, error) {
+	query := fmt.Sprintf(`
+		from(bucket: "%s")
```

Comment:
I think it is better to build this string with Builder
and probably create some help function